smart-contract/src/deploy.ts

```diff
@@ -1,39 +1,28 @@
 /* eslint-disable no-console */
 import {
   Account,
-  Args,
   Mas,
   SmartContract,
   Web3Provider,
 } from '@massalabs/massa-web3';
-import { getScByteCode } from './utils';
+import { getByteCode } from './utils';
 
-async function deploy() {
-  const account = await Account.fromEnv();
-  const provider = Web3Provider.newPublicBuildnetProvider(account);
-
-  console.log('Deploying contract...');
-
-  const byteCode = getScByteCode('build', 'main.wasm');
-  const constructorArgs = new Args().addString('Massa');
-
-  const contract = await SmartContract.deploy(
-    provider,
-    byteCode,
-    constructorArgs,
-    { coins: Mas.fromString('0.01') },
-  );
-
-  console.log('Contract deployed at:', contract.address);
-
-  const events = await provider.getEvents({
-    smartContractAddress: contract.address,
-    isFinal: true,
+async function deploy(provider: Web3Provider): Promise<SmartContract> {
+  const byteCode = getByteCode('build', 'deweb-interface.wasm');
+  return await SmartContract.deploy(provider, byteCode, undefined, {
+    // TODO: Calculate the amount of coins to send
+    coins: Mas.fromString('0.01'),
   });
+}
 
-  for (const event of events) {
-    console.log('Event: ', event.data);
-  }
+async function main() {
+  const account = await Account.fromEnv();
+  const provider = Web3Provider.newPublicBuildnetProvider(account);
+  const contract = await deploy(provider);
+  console.log(`Contract deployed at: ${contract.address}`);
 }
 
-deploy();
+main().catch((error) => {
+  console.error(error);
+  process.exit(1);
+});
```

Comment:
this is all useless,
in case main throws the stacktrace will be printed in console and programm will terminate.